tests/playbooks/tests_802_1x.yml

```diff
@@ -88,7 +100,80 @@
                   system_ca_certs: True
                   domain_suffix_match: example.com
         - name: "TEST: I can ping the EAP server"
-          shell: ping -c1 203.0.113.1
+          command: ping -c1 203.0.113.1
+        - import_role:
+            name: linux-system-roles.network
+          vars:
+            network_connections:
+              - name: "{{ interface }}"
+                persistent_state: absent
+                state: absent
+        - name: >-
+            TEST: 802.1x profile with unencrypted private key and ca_path
+          debug:
+            msg: "##################################################"
+        - name: Create directory for ca_path test
+          file:
+            path: "/etc/pki/tls/my_ca_certs"
+            state: directory
+            mode: 0755
+        - name: Copy cacert to ca_path
+          copy:
+            src: "cacert.pem"
+            dest: "/etc/pki/tls/my_ca_certs/cacert.pem"
+            mode: 0644
+        - name: Hash cacert
+          command: openssl x509 -hash -noout
+            -in /etc/pki/tls/my_ca_certs/cacert.pem
+          register: cacert_hash
+        - name: Add symlink for cacert
+          file:
+            state: link
+            path: "/etc/pki/tls/my_ca_certs/{{ cacert_hash.stdout }}.0"
+            src: cacert.pem
+        - name: Get NetworkManager version
+          command:
+            cmd: rpm -qa NetworkManager
+            warn: false
+          register: __network_NM_NVR
+        - block:
+            - import_role:
+                name: linux-system-roles.network
+              vars:
+                network_connections:
+                  - name: "{{ interface }}"
+                    interface_name: veth2
+                    state: up
+                    type: ethernet
+                    ip:
+                      address:
+                        - 203.0.113.2/24
+                      dhcp4: "no"
+                      auto6: "no"
+                    ieee802_1x:
+                      identity: myhost_capath
+                      eap: tls
+                      private_key: /etc/pki/tls/client.key.nocrypt
+                      client_cert: /etc/pki/tls/client.pem
+                      private_key_password_flags:
+                        - not-required
+                      ca_path: /etc/pki/tls/my_ca_certs
+            - fail:
+                name: Check if role should have failed
```

Comment:
The ansible logs show this error:
```
TASK [fail] ********************************************************************
task path: /tmp/tmpf1of1lym/tests/playbooks/tests_802_1x.yml:161
fatal: [/cache/CentOS-7-x86_64-GenericCloud-1907.qcow2c]: FAILED! => {"changed": false, "msg": "Invalid options for fail: name"}
```
From the ansible doco it looks like the "name" option is not supported https://docs.ansible.com/ansible/latest/modules/fail_module.html

Also I don't think this will properly check if the ca_path test silently fails. Pretty sure if this "fail" tasks runs, the playbook will still run the "rescue" section. And because the "fail" task in the rescue section has opposite conditions to the "fail" task in the "block" it will never report a failure for the "ca_path test did not fail but {{ __network_NM_NVR.stdout }} known to be broken" case

Reply:
yes, good catch, thanks!

Reply:
I will have to add something like discussed here: https://github.com/linux-system-roles/network/pull/200